### PR TITLE
bug: Fixing grad accumulation

### PR DESCRIPTION
**Pull request by isamu-isozaki**
In the current logic, I believe we are just scaling the loss and are not accumulating the gradients. This should fix this. But I will double-check

### CODE REVIEW

File: src/refiners/training_utils/trainer.py [hunk 1]
```diff
@@ -346,18 +346,19 @@ def backward(self) -> None:
...
-            self.optimizer.step()
-            self.optimizer.zero_grad()
-            self._call_callbacks(event_name="on_optimizer_step_end")
-        if self.clock.is_lr_scheduler_step:
-            self._call_callbacks(event_name="on_lr_scheduler_step_begin")
-            self.lr_scheduler.step()
-            self._call_callbacks(event_name="on_lr_scheduler_step_end")
-        if self.clock.is_evaluation_step:
-            self.evaluate()
+        if self.global_step % self.clock.num_step_per_iteration == 0:
```
Comment (limiteinductive) [inline src/refiners/training_utils/trainer.py:349]: Isn't that what self.clock.is_optimizer_step is checking?

File: src/refiners/training_utils/trainer.py [hunk 2]
```diff
@@ -369,12 +370,14 @@ def step(self, batch: Batch) -> None:
 
     def epoch(self) -> None:
         """Perform a single epoch."""
+        self.global_step = 1
```
Comment (limiteinductive) [inline src/refiners/training_utils/trainer.py:373]: Isn't this the same as sefl.clock.epoch?

File: src/refiners/training_utils/trainer.py (hunk 1 above)
Comment (isamu-isozaki) [reply src/refiners/training_utils/trainer.py:349]: ah, you are right. I think this pr is already implemented then. Closing now
